### ISSUE TITLE
Data Source API does not mention needing Schema API

### ISSUE THREAD

**Issue by cfjedimaster** [gridsome/gridsome#954]
When reading the Data Store API (https://gridsome.org/docs/data-store-api/) it talks about adding custom collections and data with custom fields. However, custom fields will not work unless you add a schema first. While that is the next section in the documentation, it is natural for a developer to read this section and attempt to test it out. (Like I did.) This particular page should immediately mention the need to define the schema as soon as a custom field is used.

**Comment by hjvedvik** [comment]
I've looked at the code in https://github.com/gridsome/gridsome/issues/442#issuecomment-576910365. Gridsome won't be able to infer a schema for your collection because the nodes are added too late in the process. You need to await the `glob` method or use `glob.sync` instead.

**Comment by cfjedimaster** [comment]
Boom, thank you. I was literally about to update this because I had an earlier demo where it DID work w/o the Schema call and I was confused. Now it makes sense. I think it's a mistake that could be made easily though and the docs should mention it perhaps?

**Comment by hjvedvik** [comment]
Yes, it could be mentioned in the docs :) And Gridsome should probably show a warning in the terminal if nodes are added after the `loadSource` step while bootstrapping.
